### PR TITLE
Add a tensor differentiation runtime test

### PR DESCRIPTION
**Pull request by rxwei**
Although the AD pass isn't checked in yet, let's start adding runtime test for the existing infra so that we can make sure things work well with tensor deabstraction.

This includes a simple test that validates a recent fix: #17468.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

File: test/TensorFlowRuntime/tensor_autodiff_runtime.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import TensorFlow
+import StdlibUnittest
+import TensorFlowUnittest
+
+var TensorADTests = TestSuite("TensorAD")
+
+TensorADTests.testAllBackends("SimpleAdjointCall") {
+  let adjPlus = #adjoint(Tensor<Float>.+)
+  let x = Tensor<Float>(1)
+  expectNearlyEqual(1, adjPlus(x, x, x + x, x).scalarized())
```
Comment (dan-zheng) [inline test/TensorFlowRuntime/tensor_autodiff_runtime.swift]: Doesn't `adjPlus(x, x, x + x, x)` return a tuple?

Comment (rxwei) [reply test/TensorFlowRuntime/tensor_autodiff_runtime.swift]: Good catch

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Review by dan-zheng — APPROVED** [review]
LGTM!

**Pull request merged**